catalyst/dl/scripts/trace.py

```diff
@@ -0,0 +1,125 @@
+import argparse
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Dict, Mapping
+
+import safitty
+import torch
+from torch.utils.data.dataloader import default_collate
+
+from catalyst.dl.scripts.utils import import_experiment_and_runner
+from catalyst.dl.experiments import Experiment, Runner
+
+
+class _ForwardOverrideModel(torch.nn.Module):
+    """
+    Model that calls specified method instead of forward
+
+    (Workaround, sigle method tracing is not supported)
+    """
+    def __init__(self, model, method_name):
+        super().__init__()
+        self.model = model
+        self.method = method_name
+
+    def forward(self, *args, **kwargs):
+        return getattr(self.model, self.method)(*args, **kwargs)
+
+
+class _TracingModelWrapper(torch.nn.Module):
+    """
+    Wrapper that traces model with batch instead of calling it
+
+    (Workaround, to use native model batch handler)
+    """
+    def __init__(self, model, method_name):
+        super().__init__()
+        self.method_name = method_name
+        self.model = model
+        self.tracing_result = None
+
+    def __call__(self, *args, **kwargs):
+        method_model = _ForwardOverrideModel(
+            self.model, self.method_name
+        )
+
+        self.tracing_result = \
+            torch.jit.trace(
+                method_model,
+                *args, **kwargs
+            )
+
+
+def build_args(parser: ArgumentParser):
+    parser.add_argument("logdir", type=Path)
+    parser.add_argument("--method", "-m", default="forward",
+                        help="Model method to trace")
+
+    return parser
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    build_args(parser)
+    args = parser.parse_args()
+    return args
+
+
+def trace_model(
+    experiment: Experiment,
+    runner_type,
+    checkpoint_path,
+    method_name="forward"
+):
+    stage = list(experiment.stages)[0]
+
+    model = experiment.get_model(stage)
+    model.load_state_dict(torch.load(checkpoint_path)["model_state_dict"])
+    model.eval()
+
+    for p in model.parameters():
+        p.requires_grad_(False)
+
+    tracer = _TracingModelWrapper(model, method_name)
+    runner: Runner = runner_type(tracer.cpu(), torch.device("cpu"))
+
+    train_dataset = experiment.get_datasets_only(stage)["train"]
```

Comment:
we do not know datasets names, so I suggest use something like
```
datasets.key()[0]
```

Reply:
and what about get loaders, take first one and get dataset from it?

Reply:
Getting loaders was in the first version. But it was tremendously slow, due to large batch and slow worker initialization. Another option is to override those params, but Im not sure whether  it's a better option.

Reply:
Even it is slow, I dont think it is an issue - you need to trace you model only once per several training usually. But some additional logic with `get_datasets_only` will force you to think about API each experiment :) (like "why do we need this method?")

another issue, with this new method - we will need to maintain it in future versions and check that it works correctly. 

solution with loaders does'nt have such issues (btw, you always can dynamically change `num_workers` and `batch_size` for speedup), that is why it looks more general for me

Reply:
OK, this time you're right 😋